### PR TITLE
[MRG+1] Raise exception if negative height or width is passed to axes()

### LINKED ISSUE #7487: Funny things happen when a rectangle with negative width/height is passed to `axes()`
mpl 2.0b4
```
axes([.9, .9, -.8, -.8]); plot([1, 2])
```
negative-axes

Interative zooming seems to work but fails to draw the rubber-band; I suspect it is because the implementation of `drag_zoom` assumes that the axes' bbox extents are in the "standard" order:
```
            x1, y1, x2, y2 = a.bbox.extents
            x, lastx = max(min(x, lastx), x1), min(max(x, lastx), x2) # probably happens here
            y, lasty = max(min(y, lasty), y1), min(max(y, lasty), y2)
```
(and I'd guess there may be other issues elsewhere).

Not sure whether we want to do something about it (which may be as simple as reordering the bbox extents in the Axes constructor, or raising an exception if the width or height are negative), or just document that this is not supported and live with it.

### PR DESCRIPTION
**Pull request by samsontmr**
Fixes #7487

### CODE REVIEW

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -483,6 +483,9 @@ def __init__(self, fig, rect,
             self._position = rect
         else:
             self._position = mtransforms.Bbox.from_bounds(*rect)
+        x0, y0, x1, y1 = self._position.extents
+        if ((x1 - x0) < 0 or (y1 - y0) < 0):
```
Comment (anntzer) [inline lib/matplotlib/axes/_base.py]: You can use `self._position.width < 0 or self._position.height < 0`.

Comment (samsontmr) [inline lib/matplotlib/axes/_base.py]: Thanks for the tip!

File: lib/matplotlib/axes/_base.py [hunk 2]
```diff
@@ -483,6 +483,8 @@ def __init__(self, fig, rect,
             self._position = rect
         else:
             self._position = mtransforms.Bbox.from_bounds(*rect)
+        if self._position.width < 0 or self._position.height < 0:
+            raise ValueError('Height and width specified must be non-negative')
```
Comment (anntzer) [inline lib/matplotlib/axes/_base.py]: Very minor point: probably "Width and height must be non-negative" is better (as the order is x, y).  I wouldn't add "specified" but I'll let a native speaker chime in :p

Comment (samsontmr) [inline lib/matplotlib/axes/_base.py]: I'm a native speaker ;p

Comment (anntzer) [inline lib/matplotlib/axes/_base.py]: That's fine, the point about order still remains valid :p

Comment (anntzer) [inline lib/matplotlib/axes/_base.py]: (I would suggest squashing the commits too.)

Comment (samsontmr) [inline lib/matplotlib/axes/_base.py]: Got it!

**Comment by samsontmr** [comment]
Commits squashed!

**Comment by anntzer** [comment]
LGTM.

**Pull request merged**